File: backend/src/main/java/com/regroup/engine/CornerAttribute.java
package com.regroup.engine;

/** One row per symbol in gameRules.md's corner property table. */
public enum CornerAttribute {
    EMPTY(StatCategory.NONE, 0, 0, 0, 0, 0, 0),
    HP_POTION_COIN(StatCategory.NONE, 1, 1, 0, 0, 0, 0),
    COINS_2(StatCategory.NONE, 0, 2, 0, 0, 0, 0),
    PA_1(StatCategory.PA, 0, 0, 1, 0, 0, 0),
    PA_2(StatCategory.PA, 0, 0, 2, 0, 0, 0),
    PD_1(StatCategory.PD, 0, 0, 0, 1, 0, 0),
    PD_2(StatCategory.PD, 0, 0, 0, 2, 0, 0),
    MA_1(StatCategory.MA, 0, 0, 0, 0, 1, 0),
    MA_2(StatCategory.MA, 0, 0, 0, 0, 2, 0),
    MD_1(StatCategory.MD, 0, 0, 0, 0, 0, 1),
    MD_2(StatCategory.MD, 0, 0, 0, 0, 0, 2);

    private final StatCategory category;
    private final int hpp;
    private final int coins;
    private final int pa;
    private final int pd;
    private final int ma;
    private final int md;

    CornerAttribute(StatCategory category, int hpp, int coins, int pa, int pd, int ma, int md) {
        this.category = category;
        this.hpp = hpp;
        this.coins = coins;
        this.pa = pa;
        this.pd = pd;
        this.ma = ma;
        this.md = md;
    }

    public StatCategory category() {
        return category;
    }

    public int hpp() {
        return hpp;
    }

    public int coins() {
        return coins;
    }

    public int pa() {
        return pa;
    }

    public int pd() {
        return pd;
    }

    public int ma() {
        return ma;
    }

    public int md() {
        return md;
    }
}

File: mobile/lib/domain/models/corner_attribute.dart
/// Mirrors WS_CONTRACT.md's `CornerAttribute` symbols exactly (see backend/gameRules.md).
enum CornerAttribute {
  empty,
  hpPotionCoin,
  coins2,
  pa1,
  pa2,
  pd1,
  pd2,
  ma1,
  ma2,
  md1,
  md2;

  String get wireName => switch (this) {
    CornerAttribute.empty => 'EMPTY',
    CornerAttribute.hpPotionCoin => 'HP_POTION_COIN',
    CornerAttribute.coins2 => 'COINS_2',
    CornerAttribute.pa1 => 'PA_1',
    CornerAttribute.pa2 => 'PA_2',
    CornerAttribute.pd1 => 'PD_1',
    CornerAttribute.pd2 => 'PD_2',
    CornerAttribute.ma1 => 'MA_1',
    CornerAttribute.ma2 => 'MA_2',
    CornerAttribute.md1 => 'MD_1',
    CornerAttribute.md2 => 'MD_2',
  };

  /// Human-readable label, used as a tooltip/accessibility label.
  String get label => switch (this) {
    CornerAttribute.empty => 'Empty',
    CornerAttribute.hpPotionCoin => '1 HP potion + 1 coin',
    CornerAttribute.coins2 => '2 coins',
    CornerAttribute.pa1 => '1 physical attack',
    CornerAttribute.pa2 => '2 physical attack',
    CornerAttribute.pd1 => '1 physical defense',
    CornerAttribute.pd2 => '2 physical defense',
    CornerAttribute.ma1 => '1 magic attack',
    CornerAttribute.ma2 => '2 magic attack',
    CornerAttribute.md1 => '1 magic defense',
    CornerAttribute.md2 => '2 magic defense',
  };

  static CornerAttribute fromWire(Object? value) {
    return CornerAttribute.values.firstWhere(
      (a) => a.wireName == value,
      orElse: () => CornerAttribute.empty,
    );
  }
}
